### PR TITLE
Disable migration aliases in index pattern

### PR DESCRIPTION
**Pull request by ruflin**
The migration aliases should not show up in the index pattern if `migration.enabled: false`. For this to happen, the Kibana index pattern must be generated on the fly instead of packaging it with each Beat. This PR introduces the generation of the index pattern when Kibana data is loaded.

APM still needs the index pattern as file. For this the export command `index-pattern` was added. It will print the index pattern to the standard out:

```
./metricbeat export index-pattern > pattern.json
```

The commands to generate the index pattern in the dev environment were removed.

For checking if aliases are supported, the Kibana version is checked. Fully accurate would be to check the Elasticsearch version as it depends on the ES version in the end and not Kibana. But it's assume that in general the same minor version is used. The reason not Elasticsearch is checked as it would potentially require additional config options and adds unnecessary complexity.

For the index pattern the internal fields.go are used. Even if fields.yml is configured still fields.go is used. This is the same behavior as we had so far when the index pattern was generated. It could be improved in the future to also support a fields.yml for the generation if needed.

In general this PR tried to change as little code as possible. The code and tests around the Kibana dashboard generation and index pattern generation is not very nice. One reason is that it also contains old logic which was used for previous versions but also has the ability to read dashboards from a zip file. Because of this all the old capabilities have to stay in the code for now. The code should be cleaned up at a later stage.

Further changes:

* Added system tests to Filebeat to check for correct content in index pattern when migration is enabled.
* Fix double generation of common.yml in Metricbeat. It seems some changes in the past caused that the common.yml file was contained in two fields.go files in Metricbeat

### CODE REVIEW

File: libbeat/kibana/fields_transformer.go [hunk 1]
```diff
@@ -115,6 +121,7 @@ func (t *fieldsTransformer) transformFields(commonFields common.Fields, path str
 
 func (t *fieldsTransformer) update(target *common.MapStr, override common.Field) error {
 	field, _ := transformField(t.version, override)
+	return nil
```
Comment (ruflin) [inline libbeat/kibana/fields_transformer.go]: @graphaelli @simitt I get the following error if I don't skip the duplicated check:

```
Exiting: error generating index pattern: field <metricset.module> is duplicated, remove it or set 'overwrite: true', {Name:metricset.module Type:keyword Description:Name of the module this data is coming from.
This information is coming from the modules used in Beats or Logstash.
```

I'm trying to figure out why this is the case as metricbeat.module is as far as I can see not defined twice. Any idea?

Comment (simitt) [reply libbeat/kibana/fields_transformer.go]: I tried to run `make update` on metricbeat, but I cannot reproduce this error.

Comment (ruflin) [reply libbeat/kibana/fields_transformer.go]: It does not happen on `make update` but only when running `./metricbeat export index-template`. That is the strange thing (or probably the key hint why it does not work)

Comment (simitt) [reply libbeat/kibana/fields_transformer.go]: It looks like everything defined in the `metricbeat/_meta/fields.common.yml` is also defiined in the created `libbeat/fields.yml`, and both of them are used for the index pattern creation.

File: CHANGELOG-developer.next.asciidoc [hunk 2]
```diff
@@ -21,6 +21,7 @@ The list below covers the major changes between 7.0.0-alpha2 and master only.
 ==== Breaking changes
 - Outputs receive Index Manager as additional parameter. The index manager can
   be used to create an index selector. {pull}10347[10347]
+- Remove support for loading dashboards to Elasticsearch 5. {pull}10451[10451]
```
Comment (ruflin) [inline CHANGELOG-developer.next.asciidoc]: This was missing in a previous PR.

**Pull request merged**